posydon/binary_evol/CE/step_CEE.py

```diff
@@ -397,6 +397,487 @@ def calculate_lambda_CE(self, donor, verbose=False):
+            core mass of the donor (in Msun)
+        rc1_i : float
+            core radius of the donor (in Rsun)
+        comp_star : SingleStar object
+            The companion star
+        mc2_i : float
+            core mass of the companion (in Msun)
+        rc2_i : float
+            core radius of the companion (in Rsun)
+        separation_postCEE : float
+            binary's separation upon exiting the CEE (in cm)
+        verbose : bool
+            In case we want information about the CEE.
+
+        Returns
+        -------
+        m1c_f: float
+            donor mass after the CE (in Msun)
+        r1c_f: float
+            donor radius after the CE (in Rsun)
+        m2c_f: float
+            companion mass after the CE (in Msun)
+        r2c_f: float
+            companion radius after the CE (in Rsun)
+        separation_f : float
+            binary's final separation upon exiting the CE (in Rsun)
+        orbital_period_f : float
+            binary's final orbital period upon exiting the CE (in days)
+        merger : bool
+            whether the binary merged in the CE
+        """
+        # First find the post-CE parameters for each star
+        mc1_f, rc1_f, mc2_f, rc2_f = self.CEE_adjust_post_CE_core_masses(
+            donor, mc1_i, rc1_i, donor_type, comp_star, mc2_i, rc2_i,
+            comp_type, double_CE, verbose=verbose)
+
+        # calculate the orbital period after CEE (before adjustment)
+        orbital_period_postCEE = cf.orbital_period_from_separation(
+            separation_postCEE / const.Rsun, mc1_i, mc2_i)
+
+        # First, check if merger happens before stable MT phase. Note this uses
+        # the 1% core values to calculate the He-core radii, but it uses the 
+        # post-CEE masses
+        merger = cf.check_for_RLO(mc1_i, rc1_f, mc2_i, rc2_f,
+            separation_postCEE / const.Rsun, self.CEE_tolerance_err)
+
+        if merger:
+            if verbose:
+                print("system merges within the CEE, prior to stable MT")
+            return mc1_i, rc1_f, mc2_i, rc2_f, \
+                separation_postCEE / const.Rsun, orbital_period_postCEE, merger
+        else:
+            if verbose:
+                print("system survives initial CEE detachment")
+                print("donor core mass / core radius:", mc1_f, rc1_i)
+                print("companion core mass / core radius:", mc2_f, rc2_i)
+
+
+        # An assumed wind loss after postCEE with mass lost
+        # from the vicinity of the donor
+        orbital_period_f = cf.period_change_stable_MT(
+            orbital_period_postCEE, Mdon_i=mc1_i, Mdon_f=mc1_f,
+            Macc_i=mc2_i, alpha=1.0, beta=0.0)
+
+        if double_CE:
+            # a wind mass loss from the 2nd star assumed to be
+            # happening at the same time
+            orbital_period_f = cf.period_change_stable_MT(
+                orbital_period_f, Mdon_i=mc2_i, Mdon_f=mc2_f,
+                Macc_i=mc1_i, alpha=1.0, beta=0.0)
```

Comment:
I think this is a bug, which was in the old code already. In the case of pure wind mass loss the period change simplifies to period_f = period_i * ((M1_i+M2_i)/(M1_f+M2_f))**2. We get this in case we use in the second call `Macc_i=mc1_f` as period_f = period_i * ((mc1_i+mc2_i)/(mc1_f+mc2_i))**2 * ((mc2_i+mc1_f)/(mc2_f+mc1_f))**2 = period_i * ((mc1_i+mc2_i)/(mc1_f+mc2_f))**2
```suggestion
            # a wind mass loss from the 2nd star assumed to be
            # happening at the same time
            orbital_period_f = cf.period_change_stable_MT(
                orbital_period_f, Mdon_i=mc2_i, Mdon_f=mc2_f,
                Macc_i=mc1_f, alpha=1.0, beta=0.0)
```